### DISCUSSION TITLE
Preview Build Fails in Swift Package

### DISCUSSION THREAD

**Discussion by ParkTaeGyu** [pointfreeco/swift-composable-architecture#2706] [General]
### Description

I'm designing a multimodule project by organizing local packages myself.
I'm in the process of isolating a `UI` module and noticed that the Preview feature is not available inside that package.
Unusually, the preview feature is available in other modules that import the `UI` module.

### Checklist

- [X] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-composable-architecture/issues) or [discussion](https://github.com/pointfreeco/swift-composable-architecture/discussions).

### Expected behavior

Run Preview Success inside `UI` Package

### Actual behavior

Build Preview Fails

### Steps to reproduce

1. Make Local Package in Xcode
2. Edit Package.swift  
<img width="1114" alt="image" src="https://github.com/pointfreeco/swift-composable-architecture/assets/13725655/82e6c00b-41aa-4130-9c79-d4b3a8f0d7fa">

3. In Package, Create SwiftUI View 
<img width="438" alt="image" src="https://github.com/pointfreeco/swift-composable-architecture/assets/13725655/ba6acb02-4b9a-4ae1-b0d5-78d37dba0322">

4. Run Preview
5. Preview Build Fails in My Package. My Package name is UI.
<img width="1103" alt="image" src="https://github.com/pointfreeco/swift-composable-architecture/assets/13725655/4b2bed99-415f-43e6-89f0-2349af2cf51f">

### The Composable Architecture version information

1.5.0

### Destination operating system

iOS 17

### Xcode version information

Version 15.0 (15A240d)

### Swift Compiler version information

```shell
swift-driver version: 1.87.1 Apple Swift version 5.9 (swiftlang-5.9.0.128.108 clang-1500.0.40.1)
Target: arm64-apple-macosx14.0
```

**Comment by mbrandonw** [comment]
Hi @ParkTaeGyu, can you provide a minimal project that demonstrates the problem you are seeing? I don't think this is an issue with the library, and most likely an issue with the project configuration or Xcode previews, and so I am going to convert it to a discussion.
